lib/jsxgettext.js

```diff
@@ -237,16 +268,16 @@ function parse(sources, options) {
       return item && arr.indexOf(item) === i;
     }
 
-    Object.keys(translations).forEach(function (msgid) {
-      var comments = translations[msgid].comments;
-
-      if (!comments)
-        return;
-
-      if (comments.reference)
-        comments.reference = comments.reference.split('\n').filter(dedupeNCoalesce).join('\n');
-      if (comments.extracted)
-        comments.extracted = comments.extracted.split('\n').filter(dedupeNCoalesce).join('\n');
+    Object.keys(translations).forEach(function (msgctxt) {
+      Object.keys(translations[msgctxt]).forEach(function (msgid){
```

Comment:
How about you extract the inner function definition to outside of the `forEach` iterations and then pass `msgctxt` as the "context" argument to the inner `forEach` iteration? This way you can avoid redefining the inner function for each context iteration.